homeassistant/components/lamarzocco/binary_sensor.py

```diff
@@ -0,0 +1,95 @@
+"""Binary Sensor platform for La Marzocco espresso machines."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from lmcloud import LMCloud as LaMarzoccoClient
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import LaMarzoccoEntity, LaMarzoccoEntityDescription
+
+
+@dataclass(frozen=True, kw_only=True)
+class LaMarzoccoBinarySensorEntityDescription(
+    LaMarzoccoEntityDescription,
+    BinarySensorEntityDescription,
+):
+    """Description of a La Marzocco binary sensor."""
+
+    is_on_fn: Callable[[LaMarzoccoClient], bool]
+    icon_on: str
+    icon_off: str
+
+
+ENTITIES: tuple[LaMarzoccoBinarySensorEntityDescription, ...] = (
+    LaMarzoccoBinarySensorEntityDescription(
+        key="water_tank",
+        translation_key="water_tank",
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        icon_on="mdi:water-remove",
+        icon_off="mdi:water-check",
+        is_on_fn=lambda lm: not lm.current_status.get("water_reservoir_contact"),
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    LaMarzoccoBinarySensorEntityDescription(
+        key="brew_active",
+        translation_key="brew_active",
+        device_class=BinarySensorDeviceClass.RUNNING,
+        icon_off="mdi:cup-off",
+        icon_on="mdi:cup-water",
+        is_on_fn=lambda lm: bool(lm.current_status.get("brew_active")),
+        available_fn=lambda lm: lm.websocket_connected,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up binary sensor entities."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities: list[LaMarzoccoBinarySensorEntity] = []
+    for description in ENTITIES:
+        if coordinator.lm.model_name in description.supported_models:
+            if (
+                description.key == "brew_active"
+                and not coordinator.local_connection_configured
+            ):
+                continue
```

Comment:
Instead of this code, I would suggest adding some sort of "supported" function to the entity description, which returns a boolean if it is supported or not.
With that method we could also remove the other if statement too